paasta_tools/deployd/leader.py

```diff
@@ -54,5 +54,14 @@ def reconnection_listener(self):
             time.sleep(5)
             attempts += 1
         self.log.error("Connection did not recover, abdicating!")
+        self.suicide()
+
+    def suicide(self):
```

Comment:
can we name this something else? ``def terminate`` should be ok (maybe even ``_terminate`` to indicate this isn't really a public api)